js/newsList.js

```diff
@@ -0,0 +1,216 @@
+  document.querySelector(".left-btn").addEventListener("click", () => clickListLeftBtn(now_category));
+}
+
+function nextNewsWhenProgressEnd() {
+  if (page_count[now_category] === total_pages[now_category] - 1) {
+    if (checkLastCategory()) {
+      setFisrtCategory();
+    } else {
+      page_count[now_category] = 0;
+      const $progress_bar = document.querySelector(".progress-bar");
+      removeProgressIterEvent($progress_bar);
+      now_category = category[category.indexOf(now_category) + 1];
+      removeProgressStatus();
+      document.querySelectorAll(".nav-item").forEach(nav => {
+        if (nav.textContent === now_category) {
+          const $nav_li = nav.parentElement;
+          $nav_li.classList.add("progress-bar");
+          insertCountDiv($nav_li);
+          addProgressIterEvent($nav_li);
+        }
+      });
+    }
+  } else {
+    page_count[now_category] += 1;
+  }
+  redrawNewsContents();
+}
+
+function redrawNewsContents() {
+  drawNews(now_category, page_count[now_category]);
+  setNowCount();
+  drawArrow();
+}
+
+function insertCountDiv(component) {
+  component.insertAdjacentHTML(
+    "beforeend",
+    `
+  <div class="count font-init"><span class="now-count">${page_count[now_category] + 1}</span> <span>/</span><span class="total-count">${
+      total_pages[now_category]
+    }</span></div>
+  `,
+  );
+}
+
+function addProgressIterEvent(component) {
+  component.addEventListener("animationiteration", nextNewsWhenProgressEnd);
+}
+
+function removeProgressIterEvent(component) {
+  component.removeEventListener("animationiteration", console.log("erase!"));
+}
+
+function removeProgressStatus() {
+  document.querySelector(".progress-bar").classList.remove("progress-bar");
+  document.querySelector(".count").remove();
+}
+
+function setNowCount() {
+  document.querySelector(".now-count").textContent = page_count[now_category] + 1;
+}
+
+function hideArrow(direction) {
+  document.querySelector(`.${direction}-btn`).classList.add("hidden");
+}
+
+function showArrow(direction) {
+  document.querySelector(`.${direction}-btn`).classList.remove("hidden");
+}
+
+function checkSameCategory(target) {
+  if (now_category === target.firstElementChild.textContent.trim()) {
+    return;
+  }
+}
+
+function checkLastCategory() {
+  if (now_category === category[category.length - 1] && page_count[now_category] === total_pages[now_category] - 1) {
+    return true;
+  }
+  return false;
+}
+
+function setFisrtCategory() {
+  now_category = category[0];
+  const $first_category = document.querySelector(".nav-item").parentElement;
+  clickCategory($first_category);
+}
+
+function setSubListNav() {
+  const subscribed_presses = presses.filter(press => press.isSub === true);
+  console.log(subscribed_presses);
+  const $sub_list_nav = document.querySelector(".sub-list-nav").firstElementChild;
+  [...subscribed_presses].forEach(press => {
```

Comment:
```suggestion
  subscribed_presses.forEach(press => {
```
filter은 반환값이 Array라 굳이 spread operator 작성안하셔도 될 것 같아요